py/ngff_zarr/v06/zarr_metadata.py

```
@@ -0,0 +1,426 @@
+                    transformations=sub_transforms
+                )
+            else:
+                raise ValueError(f"Unsupported transform type: {transform['type']}")
+            
+            # resolve input/output to CoordinateSystem instances if matching name found
+            # because coordinate system may not exist for multiscale transforms
+            # where input is a dataset path
+            coordinate_system_names = [cs.name for cs in coordinateSystems]
+            input = transform.get("input", None)
+            if input in coordinate_system_names:
+                input = [cs.name for cs in coordinateSystems if cs.name == input.name][0]
```

Ruff (0.15.17)

[warning] Prefer `next(...)` over single element slice

Replace with `next(...)`

(RUF015)